### PR TITLE
fix: value too long error

### PR DESCRIPTION
**Pull request by billgeo**
https://github.com/linz/ds-infra-team/issues/864

Alternative to https://github.com/linz/linz-bde-schema/pull/218

### CODE REVIEW

File: sql/04-lds_layer_functions.sql.in [hunk 1]
```diff
@@ -2533,7 +2533,7 @@ BEGIN
         PRPT.char_value::VARCHAR(10) AS owner_type,
         PRP.prime_surname,
         PRP.prime_other_names,
-        PRP.corporate_name,
+        PRP.corporate_name::TEXT AS corporate_name,
```
Comment (palmerj) [inline sql/04-lds_layer_functions.sql.in:2536]: Don't need the alias here, it's superfluous as the source column has the same output name

File: sql/90-patches.sql [hunk 2]
```diff
@@ -134,6 +134,64 @@ $$
...
 );
 
+PERFORM _patches.apply_patch(
+    'lds - 1.14.0: Use `text` field for anything depending on `LDS_GetProtectedText`',
+    $PATCH$
+DO $$
+DECLARE
+    versioning_enabled BOOL;
+    table_is_versioned BOOL;
+    columns schema_table_column[] := ARRAY([
```
Comment (palmerj) [inline sql/90-patches.sql:144]: @l0b0 you could simplify this array defintion and loop to something like:
```
DO $$
DECLARE
    schema_name TEXT;
    table_name TEXT;
    column_name TEXT;
BEGIN
    FOR schema_name, table_name, column_name IN
          VALUES
            ('bde_ext', 'alias', 'surname'),
            ('bde_ext', 'nominal_index', 'corporate_name'),
            ('bde_ext', 'proprietor', 'corporate_name'),
            ('lds', 'title_owners', 'owner'),
            ('lds', 'title_owners_aspatial', 'corporate_name'),
            ('lds', 'titles_plus', 'owners')
    LOOP
        RAISE NOTICE 'Schema %, table %, column %', schema_name, table_name, column_name;
    END LOOP;
END;
$$
```

Then you don't need to define and drop the TYPE that you are defining.

**Review by billgeo — COMMENTED** [review]
Also, doc/lds-full-landonline-data-dictionary-and-models.md and doc/property-and-ownership-simplified-tables-data-dictionary.md have a version number that needs to be bumped and new changelog entry added.

https://github.com/linz/linz-lds-bde-schema/blob/d6976bac6e65868ab16dd3adc14de38a3dac33cb/doc/lds-full-landonline-data-dictionary-and-models.md?plain=1#L4-L18

https://github.com/linz/linz-lds-bde-schema/blob/d6976bac6e65868ab16dd3adc14de38a3dac33cb/doc/property-and-ownership-simplified-tables-data-dictionary.md?plain=1#L4-L17

**Comment by billgeo** [comment]
I think we need to avoid use of the word `protected`, for security reasons. And also the message is probably a bit too specific and technical for the intended audience.

Could change `Use unlimited length for protected text columns` to `Minor schema changes` or `Table column definition changes`, if we want to be a bit more specific.

**Comment by iharrison** [comment]
Does https://github.com/linz/linz-lds-bde-schema/blob/49ca935c82fb49e45f894f93d13eeed3f1c3be2e/sql/05-bde_ext_schema.sql#L88 also have to be updated or does patches.sql fix this?

**Comment by billgeo** [comment]
>  or does patches.sql fix this?

yes, the patch will be applied and change it from varchar(250) to text.

**Review by iharrison — APPROVED** [review]

**Review by billgeo — COMMENTED** [review]
LGTM

**Comment by palmerj** [comment]
Didn't get to this review in time before in was merged. However I think we have missed dealing with `name VARCHAR(255)` column in [bde_ext.encumbrancee ](https://github.com/linz/linz-lds-bde-schema/blob/master/sql/05-bde_ext_schema.sql#L106). There is also an overflow character risk because of [LDS_GetEncumbranceeName](https://github.com/linz/linz-lds-bde-private/blob/master/sql/01-lds_protect_functions.sql.in#L49). So we also change the type for the name column to TEXT in bde_ext.encumbrancee as the [original source table](https://github.com/linz/linz-bde-schema/blob/master/sql/02-bde_schema.sql#L557) allows up to 255 as well.

**Comment by l0b0** [comment]
@iharrison No, only the ones which actually receive the output of the relevant function, which I believe we've tracked down.

**Comment by l0b0** [comment]
@palmerj How would overflow be triggered, though? If `name` is `varchar(255)`, and we either return it verbatim or instead a fixed string `under the Land Transfer Act 2017`, there doesn't seem to be any way for that function to return anything longer than 255 characters.

**Pull request merged**